src/Install.php

```diff
@@ -278,10 +278,35 @@ public static function get_db_update_callbacks() {
 		return self::$db_updates;
 	}
 
+	/**
+	 * Is a DB update needed?
+	 *
+	 * @return boolean
+	 */
+	public static function needs_db_update() {
```

Comment:
This processes the updates but doesn't update the DB version so it ends up being called on every dashboard load.

Reply:
I think thats the way its supposed to work as this is the portion that contains the check.

Reply:
Sorry, I should have included the db version I was testing with

```
$ wp option get wc_admin_version
0.19.0
```

I added some debug code to verify that `needs_db_update()` runs on every dashboard load. `wc_admin_version` doesn't get updated to the current version.

Reply:
> and already have completed actions for the upgrade functions? You have to delete those in order for new ones to get scheduled.

That was it.